### PR TITLE
Feature/widget synology download

### PR DESCRIPTION
**Pull request by remygrandin**
[remygrandin](https://github.com/remygrandin) ✨ Feature Medium [remygrandin /FEATURE/widget-synology-download → Lissy93/dashy](https://github.com/Lissy93/dashy/tree/FEATURE/widget-synology-download) Commits: 5 | Files Changed: 5 | Additions: 233 Unchecked Tasks

*Thank you for contributing to Dashy! So that your PR can be handled effectively, please populate the following fields (delete sections that are not applicable)*

**Category**: 
Feature

**Overview**
Proposing a new widget displaying the download/torrent tasks from a Synology NAS

Note this include an addition of a new option into the percentageChart component to add the ability to do a chart without labels (showLegend default to true to not break backward compatibility with existing usage). I didn't find any related documentation to update.

**Screenshot**
Widget Screenshot

**Code Quality Checklist** _(Please complete)_
- [X] All changes are backwards compatible
- [X] All lint checks and tests are passing
- [X] There are no (new) build warnings or errors
- [ ] _(If a new config option is added)_ Attribute is outlined in the schema and documented (N/A)
- [ ] _(If a new dependency is added)_ Package is essential, and has been checked out for security or performance (N/A)
- [ ] Bumps version, if new feature added (N/A ?)

### CODE REVIEW

**Comment by remygrandin** [comment]
Update : I fixed the issue located by sonarcube (b tag instead of strong)

File: docs/widgets.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ Dashy has support for displaying dynamic content in the form of widgets. There a
   - [Pi Hole Queries](#pi-hole-queries)
   - [Recent Traffic](#recent-traffic)
   - [Stat Ping Statuses](#stat-ping-statuses)
+  - [Synology Download Station](#stat-ping-statuses)
```
Comment (Lissy93) [inline docs/widgets.md:49]: This should be
```
- [Synology Download Station](#synology-download-station)
```

**Review by Lissy93 — APPROVED** [review]
Awesome!
Great widget, thank you for adding :)
Code looks all good, I don't have Synology device so cannot properly test it, but I don't see any problems, nice work.

**Comment by remygrandin** [comment]
Thanks.

I tested it on my on setup, which is a synology with the latest DSM 7 OS, but as I understand the API docs from them, it should work for most of their NAS if they where updated in the last 10 years.

There 2 points where I need your input, potentially for a futur amelioration of this widget (or other one I intend to propose for other services) :
 - What is the best strategy to add icons ? I saw in your other widget you referenced assets/interface-icons/ but I wasn't sure from what pack you sourced them from. I guess the idea is to add the strict minimum needed and to try to keep them generic enough to fit in most theme and be reusable in other section/widgets ?
 - I faced an issue that the TOTP protected accounts in synology coudn't login in the API without... the OTP of course. So I only handeled basic auth right now but I plan to add this support in the futur. However, for the computation of the OTP itself, I would need to store the OTP key (easy to add in the widget config) and probably a dedicated library to handle the computation (it is not a complex hash but I would prefere to leave security computation like this to a "professional" and not copy-paste it barbarian like). What is your prefered process here. Do you whish to discuss extra library in an issue ticket beforehand or do you whish to validate it along the PR that use it ?

PS : I really liked the hardcore linter you put in you build stack. I was surprised at first but it really help to keep it clean and legible :-)

**Comment by Lissy93** [comment]
> What is the best strategy to add icons ? I saw in your other widget you referenced assets/interface-icons/ but I wasn't sure from what pack you sourced them from. I guess the idea is to add the strict minimum needed and to try to keep them generic enough to fit in most theme and be reusable in other section/widgets ?

Yup, `@/assets/interface-icons/` is the correct location. I think most of the icons were sourced from [font awesome](https://fontawesome.com/icons), but so long as they're in an SVG format, then any thing is fine

(unless you're talking about the icons by the section title in the widget screenshots, these are just set by the user in the conf.yml file under section.icon and are the same as item icons)


> I faced an issue that the TOTP protected accounts in synology coudn't login in the API without... the OTP of course

Let's get this one merged, then discuss 2FA in a separate PR...

It's a shame that the Synology API doesn't allow users to generate a scoped API key for accessing the service, that would be a lot easier

In the future, I'm thinking about implementing a more secure way to store secrets. As having API keys and credentials in plaintext in the conf.yml doesn't feel right (though for the time being, is probably okay when just running within your LAN)

**Comment by Lissy93** [comment]
Let me know when you're ready for this to be merged :)

**Comment by remygrandin** [comment]
Ok for the Icons, I will know that for futur additions.

 > It's a shame that the Synology API doesn't allow users to generate a scoped API key for accessing the service, that would be a lot easier

Indeed, even more that they don't have a proper setting for cors, forcing to go through the cors-proxy... In addition the connection user need to be admin on the nas or it will only see his own downloads, that why protecting it with totp will be necessary. I will do a V2 of this in the comming weeks with that, as you said we can discuss that then in the dedicated PR.

👍  On my side this is ready to be merged :-)

**Review by Lissy93 — APPROVED** [review]

**Comment by Lissy93** [comment]
Merged 🎉 Usually takes about 20 minutes for the Docker automation to publish to DH, but after that it should be available.
Nice work on this btw :)

**Pull request merged**